### PR TITLE
Added invoice download APIs. Added billing account-level invoice APIs…

### PR DESCRIPTION
**Pull request by jbuls**
- Added missing invoice download APIs.
- Added billing account-level invoice APIs for supporting EA invoices.  
- Added documentNumber property to the Document type, so each EA invoice download URL has a corresponding documentNumber.  This is to support the scenario where multiple PDFs documents are generated for a single EA invoice because the invoice contains many line items.  

------------------ 

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [ ] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json [hunk 1]
```diff
@@ -1604,7 +1604,7 @@
           }
         },
         "operationId": "Invoices_ListByBillingAccount",
-        "description": "Lists the invoices for a billing account for a given start date and end date. The operation is supported for billing accounts with agreement type Microsoft Partner Agreement or Microsoft Customer Agreement.",
+        "description": "Lists the invoices for a billing account for a given start date and end date. The operation is supported for billing accounts with agreement type Microsoft Partner Agreement or Microsoft Customer Agreement or Enterprise Agreement.",
```
Comment (amberbhargava) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json:1607]: type Microsoft Partner Agreement, Microsoft Customer Agreement or Enterprise Agreement."

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json [hunk 2]
```diff
@@ -1643,6 +1643,45 @@
...
+        "tags": [
+          "Invoices"
+        ],
+        "x-ms-examples": {
+          "EnterpriseAgreementInvoice": {
+            "$ref": "./examples/BillingAccountInvoice.json"
+          }
+        },
+        "operationId": "Invoices_GetBillingAccountInvoice",
+        "description": "Gets an invoice by billing account name and ID. The operation is supported for billing accounts with agreement type Microsoft Partner Agreement or Microsoft Customer Agreement or Enterprise Agreement.",
```
Comment (amberbhargava) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json:1657]: type Microsoft Partner Agreement, Microsoft Customer Agreement or Enterprise Agreement.",

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json [hunk 3]
```diff
@@ -6359,6 +6398,11 @@
           "description": "The ID of the subscription for which the invoice is generated.",
           "type": "string",
           "readOnly": true
+        },
+        "rebillDetails": {
+          "description": "Details of an Enterprise Agreement invoice's rebill relationship to other invoices.",
```
Comment (amberbhargava) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json:6403]: I guess we can simplify it - Rebill details of an Enterprise Agreement invoice.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json [hunk 4]
```diff
@@ -6402,6 +6446,34 @@
         }
       }
     },
+    "InvoiceRebillDetails": {
+      "description": "The properties of a payment.",
```
Comment (amberbhargava) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json:6450]: The description seems wrong - "The rebill details of an invoice"

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json [hunk 5]
```diff
@@ -6402,6 +6446,34 @@
...
+            "Rebill",
+            "Original"
+          ],
+          "x-ms-enum": {
+            "name": "rebillDocumentType",
+            "modelAsString": true
+          }
+        },
+        "rebilledInvoiceId": {
+          "description": "The ID of the rebilled invoice in a Enterprise Agreement rebilled invoice relationship",
```
Comment (amberbhargava) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json:6466]: The ID of the invoice generated when an Enterprise Agreement invoice was rebilled,

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json [hunk 6]
```diff
@@ -6402,6 +6446,34 @@
         }
       }
     },
+    "InvoiceRebillDetails": {
+      "description": "The properties of a payment.",
+      "properties": {
+        "rebillDocumentType": {
+          "description": "The document type of the invoice in a Enterprise Agreement rebilled invoice relationship.",
```
Comment (amberbhargava) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json:6453]: The type of the document generated when an Enterprise Agreement invoice was rebilled,

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json [hunk 7]
```diff
@@ -6402,6 +6446,34 @@
...
+            "modelAsString": true
+          }
+        },
+        "rebilledInvoiceId": {
+          "description": "The ID of the rebilled invoice in a Enterprise Agreement rebilled invoice relationship",
+          "type": "string",
+          "readOnly": true
+        },
+        "latestInvoiceId": {
+          "description": "The ID of the latest invoice in a Enterprise Agreement rebilled invoice relationship.",
```
Comment (amberbhargava) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json:6471]: The ID of the latest invoice generated when an Enterprise Agreement invoice is rebilled,

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json (hunk 3 above)
Comment (jbuls) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json]: I expect that this should work for modern as well

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json (hunk 7 above)
Comment (jbuls) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json]: This is not specific to the multiple PDF per invoice scenario.  This is for rebill scenarios.  It references the latest generated invoice.  In a rebill scenario, there is a sequence of invoices, where the earliest is the originally-generated invoice, and the latest is the most recently generated rebill invoice.  The latest rebill invoice is the one that would be shown in the invoice list experience, and all of the earlier invoices in the sequence would be shown in the Rebill Details view for that one.

File: specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json [hunk 8]
```diff
@@ -1643,6 +1643,45 @@
         }
       }
     },
+    "/providers/Microsoft.Billing/billingAccounts/{billingAccountName}/invoices/{invoiceName}": {
```
Comment (wilcobmsft) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2019-10-01-preview/billing.json:1646]: We now have /billingAccounts/x/invoices/{invoiceName} *and* /billingAccounts/x/billingProfiles/y/invoices/{invoiceName}, and a caller will need to figure out which one to call. Really awkward, but I understand how we got here for our 2019 preview API. In our 2020 GA APIs this is all converged and makes much more sense.

**Pull request merged**